R/ModelDiagram.R

```diff
@@ -0,0 +1,135 @@
+#' Model Diagram
+#'
+#' Functions described below are helper functions for drawing
+#' diagrams from model data. These generate mermaid diagrams from
+#' transition probability matrices which can then be rendered
+#' using other packages.
+#'
+#' @name epiworld-model-diagram
+#' @examples
+#' # Create and run a model
+#' model <- ModelSIRCONN(
+#'   name = "A Virus",
+#'   n = 10000,
+#'   prevalence = .01,
+#'   contact_rate = 4.0,
+#'   transmission_rate = .5,
+#'   recovery_rate = 1.0 / 7.0
+#' )
+#'
+#' verbose_off(model)
+#' run(model, ndays = 50, seed = 1912)
+#'
+#' # Draw mermaid diagrams from model data
+#' draw_mermaid_from_data(
+#'   states = get_states(model),
+#'   transition_probs = c(get_transition_probability(model))
+#' )
+#' @param states String vector. List of model states.
+#' @param transition_probs Numeric vector. Transition probability matrix
+#' @param output_file String. Path to the output file.
```

Comment:
What happens if you specify an output (and if you don't?). Always return the string, but if output file is included, then write it out (and probably message about that)